rpc_util_test.go

```diff
@@ -294,3 +298,90 @@ func BenchmarkGZIPCompressor512KiB(b *testing.B) {
 func BenchmarkGZIPCompressor1MiB(b *testing.B) {
 	bmCompressor(b, 1024*1024, NewGZIPCompressor())
 }
+
+// compressData compresses data using gzip and returns the compressed bytes.
+func compressData(data []byte) []byte {
```

Comment:
Can you pass `testing.T` as a parameter to this function and call t.Fatalf() with the error returned by `gz.Write` and `gz.Close()` if it's not `nil`?